javascript/app.js

```diff
@@ -70,53 +82,171 @@ FinancialFreedom.controller('bodyController', function($scope, $rootScope, $loca
+                    processedParams[key] = null;
+                }
+
+                else {
+                    processedParams[key] = value;
+                }
+            }
+
+            else {
+                newVal = (value === "true");
+                
+                processedParams[key] = newVal;
+            }
 
-    $scope.calc_values = {
+        });
+
+        return processedParams;
+    }
+
+    var default_calc_values = {
         amount: null,
         expense: true,
-        recurring: false
+        recurring: true,
+        useCustomVals: true
     };
 
-    $scope.$watch('calc_values', function(new_value, old_value) {
+    if (UserDataCache.userData.monthly_expenses) {
+        default_calc_values.useCustomVals = false;
+    }
 
-        $scope.cashflowLabel = $scope.calc_values.expense ? 'Reduced from' : 'Added to';
+    $scope.convert = function() {
 
         var amount = parseInt($scope.calc_values.amount);
 
         if ($scope.calc_values.amount === '' || $scope.calc_values.amount === NaN || isNaN(amount)) {
             amount = 0;
+            return;
+        }
+        
+        $scope.preconvert = false;
+
+        if ($scope.calc_values.useCustomVals) {
+
+            customVals = { // Median American values
+                monthly_income: 4328.25,
+                total_assets: 0,
+                monthly_expenses: 4111.84,
+                income_increase_rate: 0.03,
+                growth_rate: 0.08,
+                expenses_increase_rate: 0.03
+            };  
+        }
+
+        else {
+            customVals = null;
         }
 
-        dates = DollarsToTimeService.calculateDollarsToTime(amount, $scope.calc_values.expense, $scope.calc_values.recurring);
+        var newParams = {
+            amount: amount,
+            expense: $scope.calc_values.expense,
+            recurring: $scope.calc_values.recurring,
+            useCustomVals: $scope.calc_values.useCustomVals
+        };
+
+        var stateOptions = { 
+            location: true, 
+            inherit: true, 
+            relative: $state.$current, 
+            notify: false 
+        };
+
+        $state.transitionTo('dollars-to-time', newParams, stateOptions);  
+
+        outputDate(amount, customVals, newParams);
+    };
+
+    if (params.expense !== undefined) {
+        var processedParams = processParams(params);
+        $scope.calc_values = processedParams;
+    }
+
+    else {
+        $scope.calc_values = default_calc_values;
+    }
+
+    function outputDate(amount, customVals, newParams) {
+        
+        dates = DollarsToTimeService.calculateDollarsToTime(amount, $scope.calc_values.expense, $scope.calc_values.recurring, customVals);
 
         if (dates.difference) {
             $scope.dates.years = dates.difference.years;
             $scope.dates.months = dates.difference.months;
             $scope.dates.days = dates.difference.days;
         }
+    }
+
+    $scope.showYou = function() {

```

Comment:
I'm guessing this shows the user's values instead of the median american values. If so, I'd rename this function to something more explicit like `showUserValues` or something. `showYou` isn't very explanatory